internals/secrethub/run_test.go

```diff
@@ -544,32 +560,38 @@ func TestRunCommand_Run(t *testing.T) {
 		},
 		"invalid template var: start with a number": {
 			command: RunCommand{
-				envFile: "secrethub.env",
-				osStat:  osStatNotExist,
-				templateVars: map[string]string{
-					"0foo": "value",
+				environment: &environment{
+					osStat:  osStatNotExist,
+					envFile: "secrethub.env",
+					templateVars: map[string]string{
+						"0foo": "value",
+					},
+					envar: map[string]string{},
 				},
-				envar: map[string]string{},
 			},
 			err: ErrInvalidTemplateVar("0foo"),
 		},
 		"invalid template var: illegal character": {
 			command: RunCommand{
-				envFile: "secrethub.env",
-				osStat:  osStatNotExist,
-				templateVars: map[string]string{
-					"foo@bar": "value",
+				environment: &environment{
+					osStat:  osStatNotExist,
+					envFile: "secrethub.env",
+					templateVars: map[string]string{
+						"foo@bar": "value",
+					},
+					envar: map[string]string{},
 				},
-				envar: map[string]string{},
 			},
 			err: ErrInvalidTemplateVar("foo@bar"),
 		},
 		"os env secret not found": {
 			command: RunCommand{
-				osEnv:   []string{"TEST=secrethub://nonexistent/secret/path"},
 				command: []string{"echo", "test"},
 				io:      ui.NewFakeIO(),
-				osStat:  osStatNotExist,
+				environment: &environment{
+					osEnv:  []string{"TEST=secrethub://nonexistent/secret/path"},
+					osStat: osStatNotExist,
```

Comment:
With the current implementation `osStat` and `osEnv` should also be injected into the `RunCommand`.